AutomatticTracks/src/main/java/com/automattic/android/tracks/TracksClient.java

```diff
@@ -267,6 +274,8 @@ public void run() {
                                         mInsertEventsQueue.notifyAll();
                                     }
                                 }
+                            } else {
+                                lastNetworkErrorTimestamp = 0L;
```

Comment:
I'm happy that you've found a flow path that could lead to a weird behavior. 
Even if I'm not sure I've 100% understood the flow above :)
It's unlikely that the network thread process other messages immediately after an error, because it synchronizes over `mNetworkQueue` that should be empty at that point. 
`mNetworkQueue` usually contains 1 element only, with all the events that should be sent over the wire with a single REST call.

If I got it right, the problem you've mentioned could happen if the app sends 10+ events to the library, immediately after that `sendToWireThread` has written to the queue. In that case the `networkThread` will find another event in the queue and start to precess it without waiting the timeout.
I think it's a rare case, and never get into it during the tests.